impeller/entity/entity_unittests.cc

```diff
@@ -1724,13 +1724,9 @@ TEST_P(EntityTest, RuntimeEffect) {
     entity.SetContents(contents);
     bool result = contents->Render(context, entity, pass);
 
-    if (expect_dirty) {
-      EXPECT_NE(first_pipeline, pass.GetCommands().back().pipeline);
-      first_pipeline = pass.GetCommands().back().pipeline;
```

Comment:
You're going to still want this so the else condition passes.  I'd remove that resetting of first_pipeline you added too.